### PR TITLE
[Kafka Streams] Update on dev mode

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

File: docs/src/main/asciidoc/kafka-streams-guide.adoc [hunk 1]
```diff
@@ -22,6 +22,32 @@ To complete this guide, you need:
...
 
+[NOTE]
+====
+The Quarkus extension for Kafka Streams allows for very fast turnaround times during development by supporting the Quarkus Dev Mode (e.g. via `mvn compile quarkus:dev`).
+After changing the code of your Kafka Streams topology, the application will automatically be reloaded when the next input message arrives.
+
+A recommended development set-up is to have some producer which creates test messages on the processed topic(s) in fixed intervals, e.g. every second and observe the streaming application's output topic(s) using a tool such as `kafkacat`.
+Using the dev mode, you'll instantly see messages on the output topic(s) as produced by the latest version of your streaming application when saving.
+
+For the best development experience, we recommend the following configuration settings to your Kafka broker:
```
Comment (geoand) [inline docs/src/main/asciidoc/kafka-streams-guide.adoc]: `following` should perhaps be `applying`?

Comment (geoand) [reply docs/src/main/asciidoc/kafka-streams-guide.adoc]: Also do we need both `configuration` and `settings`?

Comment (gunnarmorling) [reply docs/src/main/asciidoc/kafka-streams-guide.adoc]: Good catch on "applying" missing. Added it. I think the other is fine, but feel free to change it if you prefere.

Comment (geoand) [reply docs/src/main/asciidoc/kafka-streams-guide.adoc]: Nah, just a suggestion. I don't mind either way :).

Could you please squash however so I can merge?

**Comment by gunnarmorling** [comment]
> Could you please squash however so I can merge?

I was actually hoping you could do it using the "Squash and merge" option
or whatever there's in the PR UI. I don't have the branch locally (did the
change via the web UI) :) Thanks!

**Comment by geoand** [comment]
> Could you please squash however so I can merge?
> I was actually hoping you could do it using the "Squash and merge" option or whatever there's in the PR UI. I don't have the branch locally (did the change via the web UI) :) Thanks!

Unfortunately that's not on option in Quarkus - it has been decided for the project to always create a merge-commit and github doesn't provide the ability to both squash and create a merge commit :(

**Comment by gunnarmorling** [comment]
Ok, no worries. Created #3757 and will close this one.

**Pull request closed without merging**